src/main/java/com/google/devtools/build/lib/remote/RemoteExecutionService.java

```diff
@@ -938,13 +965,6 @@ public InMemoryOutput downloadOutputs(RemoteAction action, RemoteActionResult re
       // Check that all mandatory outputs are created.
       for (ActionInput output : action.getSpawn().getOutputFiles()) {
         if (action.getSpawn().isMandatoryOutput(output)) {
-          // Don't check output that is tree artifact since spawn could generate nothing under that
-          // directory. Remote server typically doesn't create directory ahead of time resulting in
-          // empty tree artifact missing from action cache entry.
-          if (output instanceof Artifact && ((Artifact) output).isTreeArtifact()) {
```

Comment:
There could be old cache entries that don't contain empty directory. For backward compatibility, please consider not removing this. But we can update the comment to reflect this change.

Reply:
Thanks for catching this. I restored the check and only updated the comment.